### PR TITLE
meta: reduce usages of code deprecated in sequelize.js

### PR DESCRIPTION
**Pull request by WikiRik**
## Pull Request Checklist

- [X] Have you added new tests to prevent regressions?
- [X] If a documentation update is necessary, have you opened a PR to [the documentation repository](https://github.com/sequelize/website/)? 
- [X] Did you update the typescript typings accordingly (if applicable)?
- [X] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [X] Does the name of your PR follow [our conventions](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md#6-commit-your-modifications)?

## Description of Changes

Second PR that reduces references/usages of deprecated code. This PR deals with the deprecations done in sequelize.js.

It also updates the lint-staged configuration to run prettier after eslint

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -86,6 +86,9 @@
   },
   "packageManager": "yarn@4.1.0",
   "lint-staged": {
-    "*.{js,mjs,cjs,ts,mts,cts}": "eslint --fix --report-unused-disable-directives"
+    "*": [
+      "eslint --fix --report-unused-disable-directives",
```
Comment (ephys) [inline package.json:90]: should we keep the JS filter for eslint, but use `*` for prettier?

**Review by ephys — APPROVED** [review]
one note but otherwise perfect

File: package.json (hunk 1 above)
Comment (WikiRik) [reply package.json:90]: I'd rather not. When using `format:eslint` we also go through all extensions and I would like to have the same behaviour happening for lint-staged

**Review by ephys — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (ephys) [reply package.json:90]: True!

**Pull request merged**